src/ariel-os-embassy-common/src/identity.rs

```diff
@@ -43,6 +43,81 @@ pub trait DeviceId: Sized {
 
     /// The device identifier in serialized bytes format.
     fn bytes(&self) -> Self::Bytes;
+
+    /// Generates an EUI-48 identifier ("6 byte MAC address") based on the device identity.
+    ///
+    /// See `ariel_os::identity::interface_eu48` for details.
+    fn interface_eui48(&self, if_index: u32) -> [u8; 6] {
+        // Not even trying to hash for privacy: Many CPU IDs just have 32 variable bits (eg. EFM32
+        // with a 32bit timstamp in a limited range, and a 32bit factory ID, or STM32's 96 bit
+        // containing lot and wafer numbers and coordinates), and all SHA256 hashes of 2^32
+        // possibilities can just be calculated on a graphics card in an hour.
+        //
+        // We do hash the board identifier, just to be sure to have a nice and random-looking
+        // starting point.
+
+        // FIXME: Can't depend on ariel_os due to cyclic dependencies.
+        const BOARD: &str = ariel_os_utils::str_from_env_or!(
+            "CONFIG_BOARD",
+            "unknown",
+            "board name provided by the build system"
+        );
+        const BOARD_HASH: [u8; 20] = const_sha1::sha1(BOARD.as_bytes()).as_bytes();
```

Comment:
Clarified:
```patch
+        // We do hash the board identifier, just to be sure to have a nice and random-looking
+        // starting point. The sha1 function was chosen because it is widespread (enabling
+        // re-implementation of the algorithm outside to predict addresses), available in a const
+        // implementation, and because its output is large enough to spread the input over the full
+        // address; it is not expected to be secure.
```